sys-libs/libxcrypt/libxcrypt-4.4.19.ebuild

```diff
@@ -10,10 +10,24 @@ SRC_URI="https://github.com/besser82/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
 HOMEPAGE="https://github.com/besser82/libxcrypt"
 
 LICENSE="LGPL-2.1+ public-domain BSD BSD-2"
-SLOT="0/1"
 KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86"
 IUSE="+compat split-usr +static-libs system test"
 
+export CTARGET=${CTARGET:-${CHOST}}
+if [[ ${CTARGET} == ${CHOST} ]] ; then
+	if [[ ${CATEGORY/cross-} != ${CATEGORY} ]] ; then
+		export CTARGET=${CATEGORY/cross-}
+	fi
+fi
+
+is_cross() { [[ ${CHOST} != ${CTARGET} ]] ; }
+
+if is_cross ; then
+	SLOT="${CTARGET}"
+else
+	SLOT="0/1"
```

Comment:
The SLOT has to be constant within an ebuild or it’ll violate metadata invariance

Why do we actually need this? We don’t for any other packages?

Reply:
(See [PMS](https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-600007.1) and the [devmanual](https://devmanual.gentoo.org/general-concepts/portage-cache/index.html))

Reply:
Thanks for bringing up the metadata invariance and pointing to the documentation, that is a very good point. I will drop the SLOT change as it's not necessary and I think the `is_cross` check can also be simplified to something like:
```
[[ ${CATEGORY} == cross-* ]]
```
I will do more testing then come back with an update.

Reply:
No worries, tell me if you want more pointers or anything.

re detecting cross, let’s try use [toolchain.eclass](https://devmanual.gentoo.org/eclass-reference/toolchain-funcs.eclass/)’ helper tc-is-cross-compiler

Reply:
[[ ${CATEGORY} == cross-* ]] is correct but tc-is-cross-compiler to detect is_cross is not correct.
This is based on crossdev's [1] expectations. glibc/gcc/gdb ebuilds do the same thing [2].

crossdev will create an ebuild named cross-<abi>/libxcrypt e.g. cross-armv7a-linux-gnueabihf/libxcrypt. The expectation is building this package will install files to /usr/armv7a-linux-gnueabihf/. Use of CTARGET is used to install to correct location i.e. /usr/${CTARGET}.

[1] https://wiki.gentoo.org/wiki/Cross_build_environment
[2] https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-libs/glibc/glibc-2.33.ebuild#n60